feat: Add the sudo role (#2)
linux-system-roles/sudo: README.md

# Sudo

Sudo System Role

## Requirements

This role is only supported on RHEL8.1+/CentOS8.1+ and Fedora distributions. Consider reading fapolicyd documentation before setting it up.

### Collection requirements

None.

## Role Variables

> The defaults defined for this role are based on a default RHEL7.6 `/etc/sudoers` configuration.  Please check the defaults in [`defaults/main.yml`](defaults/main.yml) prior to running for OS compatibility.

Author: I put there 8.4 for now. I guess it can be still changed later.

### sudo_rewrite_default_sudoers_file

> Use role default or user defined `sudoers_files` definition, replacing your distribution supplied `/etc/sudoers` file. Useful when attempting to deploy new configuration files to the `include_directories` and you do not wish to modify the `/etc/sudoers` file. | True | boolean |

Reviewer: Please rewrite ` | True | boolean |` to a readable format like below.
```suggestion
Use role default or user defined `sudoers_files` definition, replacing your distribution supplied `/etc/sudoers` file. Useful when attempting to deploy new configuration files to the `include_directories` and you do not wish to modify the `/etc/sudoers` file.
Default: `true`
Type: `bool`
```

`commands` List of commands to apply the specification to. You can use a defined `cmnd_alias` name as well as commands. | list |  

#### Default Override user_specifications

`defaults` List of defaults to override from the main configuration | list |  
`type` Type of default to override, this affects the operator in the configuration ( host -> `@`, user -> `:`, command -> `!`, and runas -> `>`). The type field can be one of the following values: `command`, `host`, `runas`, or `user`. | string |  
`commands` Use when `type: command`. List of `cmnd_alias` names as well as commands to override specific default values.| list |  
`hosts` Use when `type: host`. List of `host_alias` names as well as individual host names to override specific default values. | list |  
`operators` Use when `type: runas`. List of `runas_alias` names as well as individual user names to override specific default values. | list |  
`users` Use when `type: user`. List of `user_alias` names as well as individual user names to override specific default values. | list |  

## Example Playbook

```yaml
---
- name: Apply a RHEL Default /etc/sudoers configuration

>  hosts: all

Reviewer: ```suggestion
  hosts: all
```

>  roles:

Reviewer: ```suggestion
  roles:
```

>  - role: linux-system-roles.sudo

Reviewer: ```suggestion
    - role: linux-system-roles.sudo
```

Reviewer: I see incorrect indentation below too, like having 3 spaces instead of 4.
